### PR TITLE
ci: overall enhancements and cosmetics improvements

### PR DESCRIPTION
**Pull request by khos2ow**
## What does this do / why do we need it?

This PR enhances Makefile by:

- adding a `help` target, and colorizing the outputs of commands
- normalize Go execution (specifically adding `CGO_ENABLED=0` to `build`)

Also it cosmetically improves CI jobs by:

- unify them under `ci`
- unify naming
- unify Go version
- add an `[ci skip]` switch (if commit message include it, CI will be skipped)

Note: for readability I've splited them into two commits, and I personally prefer to not squash merge them. But not entirely sure what will happen to our changelog if there will be a merge commit. Anyways, either way will work (or if need be I can move one to its own PR)

## Check lists

* [x] Test passed
* [x] Coding style (indentation, etc)

### CODE REVIEW

**Comment by clok** [comment]
I want to note that a "Squash and Merge" will retain the commit messages as markdown bulleted lists in the commit message of the merge. See [ae3382b7](https://github.com/git-chglog/git-chglog/commit/ae3382b7c8175be9fec858e93058c10d89b1d8ad) as an example. I find this effective at logging the commits within a squashed commit.

The other option is a "Rebase and Merge" which does what it says in the name. That would accomplish truly separating the commits on the main line branch.
image
 
Typically, I reserve "Rebase and Merge" for items that are deemed to require an atomic separation of the commits to 1) help with maintaining why work was done and 2) allow for a more discrete change set with complex features. I do not support using "Rebase and Merge" when then are 10+ commits within a PR. Just too much noise at that point.

I am open to either in this situation. Let me know your thoughts.

File: Makefile [hunk 1]
```diff
@@ -1,37 +1,93 @@
-VERSION?=$$(git describe --tags --always)
+# Build variables
+VERSION ?= $(shell git describe --tags --always)
```
Comment (clok) [inline Makefile:2]: TIL: `shell` replaces newlines with spaces. Hmm

File: Makefile [hunk 2]
```diff
@@ -1,37 +1,93 @@
...
+	@ $(MAKE) --no-print-directory log-$@
+	git add CHANGELOG.md
+	git commit -m "chore: update changelog for $(VERSION)"
 	git tag $(VERSION)
 	git push origin master $(VERSION)
+
+########
+##@ Help
+
+.PHONY: help
```
Comment (clok) [inline Makefile:68]: Man, this is cool! 🍻

File: Makefile [hunk 3]
```diff
@@ -1,37 +1,93 @@
...
+help:   ## Display this help
+	@awk \
+		-v "col=\033[36m" -v "nocol=\033[0m" \
```
Comment (clok) [inline Makefile:71]: 🎨

**Review by clok — APPROVED** [review]
I have been working with `make` for a looooong time now and today you taught me some cool and useful stuff! Thank you for this @khos2ow!

You should consider dropping some of this knowledge on https://makefiletutorial.com/. I feel like the `help` command example could be useful on that tutorial site.

**Review by mavogel — APPROVED** [review]

**Comment by khos2ow** [comment]
> I want to note that a "Squash and Merge" will retain the commit messages as markdown bulleted lists

That's great, let's go with this. I'm just gonna go on record and say I _hate_ "Rebase and Merge", it's even worse than squash and merge! lol

> I feel like the help command example could be useful on that tutorial site.

I cannot take the whole credit here, I merely have modified this a bit along the way in the past working on top of other folks modification (I can take some credit for `log-%` part of it! :wink:). But original help implementation was introduced here: https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,35 +1,57 @@
...
     paths-ignore:
-      - 'README.md'
+      - "README.md"
+
+env:
+  GO_VERSION: "1.16"
+
 jobs:
-  unit:
+  tests:
```
Comment (khos2ow) [inline .github/workflows/ci.yml:18]: Due to this rename, now required status on PR is changed, so someone needs to update the repository to reflect this or I can revert it back `s/tests/unit/`. Both are fine with me. /cc @clok @mavogel

Comment (clok) [reply .github/workflows/ci.yml:18]: I have updated the rules on the PR checks.

Comment (khos2ow) [reply .github/workflows/ci.yml:18]: Awesome, so I'm gonna squash and merge this.

**Pull request merged**